test/domains/compilerErrors/constDomain.good

```diff
@@ -0,0 +1,2 @@
+constDomain.chpl:2: error: const actual is passed to 'ref' formal 'this' of add()
+note: to formal of type DefaultAssociativeDom(int(64),true)
```

Comment:
@leekillough: Could this test be extended to include all of the routines which you locked down as part of this PR?  (unless there's similar coverage elsewhere that I missed... I didn't catch it in this PR).  

[If the compiler only generates an error after the first, then stops, adding the `--ignore-errors-for-pass` compiler flag may cause the full set to be printed.]

Reply:
Done